AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/Storage/StorageEngine+SyncRequirement.swift

```diff
@@ -49,11 +49,23 @@ extension StorageEngine {
         }
     }
 
-    private func requiresAuthPlugin() -> Bool {
+    private func requiresAuthPlugin(api: APICategoryGraphQLBehavior?) -> Bool {
         let containsAuthEnabledSyncableModels = ModelRegistry.models.contains {
             $0.schema.isSyncable && $0.schema.hasAuthenticationRules
         }
 
+        if containsAuthEnabledSyncableModels,
+            let apiCategoryAuthProviderBehavior = api as? APICategoryAuthProviderFactoryBehavior,
+            apiCategoryAuthProviderBehavior.apiAuthProviderFactory().oidcAuthProvider() != nil {
+            if tryGetAuthPlugin() != nil {
+                log.warn("""
+Detected OIDC Auth Provider & Auth Plugin Category available.
+This is not a supported use case.
+""")
```

Comment:
Talked about this offline.  We will keep this here because there could be a case where customers have added the auth plugin with cognito, but have configured the API to use OIDC.  Multi-auth is not something we currently support, and putting this log statement here will help in the event we get issues with regards to customers who end up in this advanced/complex configuration.